src/java/org/apache/cassandra/cache/ChunkCache.java

```diff
@@ -210,14 +229,20 @@ public void release()
     public ChunkCache(BufferPool pool, int cacheSizeInMB, Function<ChunkCache, ChunkCacheMetrics> createMetrics)
     {
         cacheSize = 1024L * 1024L * Math.max(0, cacheSizeInMB - RESERVED_POOL_SPACE_IN_MB);
+        cleanupExecutor = Executors.newFixedThreadPool(CLEANER_THREADS, new NamedThreadFactory("ChunkCacheCleanup"));
         enabled = cacheSize > 0;
         bufferPool = pool;
         metrics = createMetrics.apply(this);
         keysByFile = new NonBlockingIdentityHashMap<>();
         cache = Caffeine.newBuilder()
                         .maximumWeight(cacheSize)
                         .initialCapacity(INITIAL_CAPACITY)
-                        .executor(MoreExecutors.directExecutor())
+                        .executor(r -> {
+                            if (r.getClass() == PERFORM_CLEANUP_TASK_CLASS)
+                                cleanupExecutor.execute(r);
```

Comment:
I believe we can just as well change the executor for all tasks, not just for the cleanup.

The reason the direct executor was chosen here originally was that I observed situations where the cache grew unboundedly because this task did not get a chance to run (see [CASSANDRA-11452](https://issues.apache.org/jira/browse/CASSANDRA-11452)). This was a long time ago and things probably have changed, but we may want to check that under heavy load the cache does not grow too much beyond its configured size.